src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/controller/ItemGroupController.java

```diff
@@ -59,11 +61,13 @@ public ResponseEntity<Object> createItemGroup(final @RequestHeader(X_REQUEST_ID_
             if (itemGroupsService.doesItemGroupExist(itemGroupData))
                 return buildItemAlreadyExistsResponse(xRequestId, itemGroupData);
 
-            final ItemGroupData savedItemGroupData = itemGroupsService.createItemGroup(itemGroupData);
-            return buildCreateSuccessResponse(xRequestId, savedItemGroupData);
+            final ItemGroupData savedItem = itemGroupsService.createItemGroup(itemGroupData);
+            return buildCreateSuccessResponse(xRequestId, savedItem);
         }
-        catch (Exception ex) {
-            return buildErrorResponse(xRequestId, ex, itemGroupData);
+        catch(IllegalArgumentException | MongoOperationException serverFailureException) {
+            return buildServerErrorResponse(xRequestId, serverFailureException, itemGroupData, INTERNAL_SERVER_ERROR);
+        } catch(Exception errorException) {
+            return buildServerErrorResponse(xRequestId, errorException, itemGroupData, BAD_REQUEST);
```

Comment:
This still appears to be reporting an unknown error as a problem with the request. Is that necessarily the case?